src/jemalloc.c

```diff
@@ -2362,20 +2362,26 @@ je_malloc(size_t size) {
 		return malloc_default(size);
 	}
 
-	szind_t ind = sz_size2index_lookup(size);
+	/*
+	 * The code below till the branch checking the next_event threshold may
+	 * execute before malloc_init(), in which case the threshold is 0 to
+	 * trigger slow path and init.
+	 */
+	szind_t ind;
 	/*
 	 * The thread_allocated counter in tsd serves as a general purpose
 	 * accumulator for bytes of allocation to trigger different types of
 	 * events.  usize is always needed to advance thread_allocated, though
 	 * it's not always needed in the core allocation logic.
 	 */
-	size_t usize = sz_index2size(ind);
-	/*
-	 * Fast path relies on size being a bin.
-	 * I.e. SC_LOOKUP_MAXCLASS < SC_SMALL_MAXCLASS
-	 */
+	size_t usize;
+
+	sz_size2index_usize_fastpath(size, &ind, &usize);
+	assert(usize > 0 || !malloc_initialized());
```

Comment:
Can we have `assert(usize > 0 || (!malloc_initialized() && ind == 0))` here?

Also see my comment below.

Reply:
Answered below, the ind==0 cannot be guaranteed as the initialization can be partially done.